lib/std/Thread/RwLock.zig

```diff
@@ -9,6 +9,7 @@ const RwLock = @This();
 const std = @import("../std.zig");
 const builtin = @import("builtin");
 const assert = std.debug.assert;
+const testing = std.testing;
 
 pub const Impl = if (builtin.single_threaded)
     SingleThreadedRwLock
```

Comment:
All of these locks should not be `pub` as they're meant to be implementation details

Reply:
That seems outside the scope but could be done separately if you're sure it is correct. And are you sure someone wouldn't want to use the DefaultRwLock with pthreads?

Reply:
Makes sense to do it outside the PR. DefaultRwLock is an implementation that's used when pthread isn't available. 

\<tangent\>: The current pthread implementation isn't really correct either: 1) POSIX requires that the sync primitives are pinned in memory after init/first-use. This isn't required by zig's RwLock api which is a soundness hole. 2) we're not calling deinit(), even when statically initialized. While most impls dont care, this is technically a memory leak according to POSIX.

Reply:
Do you want to create an issue for those changes?

Reply:
Btw, are you planning to do more work on the concurrent locks yourself?

Reply:
eventually:tm: While it won't be soon, I can help review changes for that effort.

Reply:
I appreciate your time and expertise on this.